mwdb/core/auth.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+from enum import Enum
+
+import jwt
+
+from mwdb.core.config import app_config
+
+
+class AuthScope(Enum):
+    session = "session"
+    api_key = "api_key"
+    legacy_api_key = "legacy_api_key"
+    set_password = "set_password"
+    download_file = "download_file"
+    legacy_token = "legacy_token"
```

Comment:
I would remove the legacy entries and use separate function for validating legacy keys. Reason is contained in further comments.
```suggestion
    session = "session"
    api_key = "api_key"
    set_password = "set_password"
    download_file = "download_file"
```